File: getchara.js
const { EnkaClient } = require("enka-network-api");
const enka = new EnkaClient();
const argument  = require('./argument.json');
const fs = require('fs');

async function get(){
    let chara = {}
    let charaList = []
    let charadata = ((await enka.fetchUser(argument["uid"])).characters)
    for(let i = 0;i < charadata.length;i++){
        chara[charadata[i].characterData.name.get("jp")] = i
        charaList.push(charadata[i].characterData.name.get("jp"))
    }
    chara["chara"] = charaList
    fs.writeFile("chara.json", JSON.stringify(chara), (err) => {
        if (err) rej(err);
        if (!err) {
          console.log('JSONファイルを生成しました');
        }
    });
    console.log(chara)
}
get()

File: getchara.py
import ast
import asyncio
import json
from enkanetwork import * 
enka = EnkaNetworkAPI()

assets = Assets(lang="jp")

async def get(uid):
    async with enka:
        chara = {}
        charaList = []
        charadata = ((await enka.fetch_user_by_uid(uid)).characters)
    for i in range(len(charadata)):
        chara[charadata[i].name] = i
        charaList.append(charadata[i].name)
    chara["chara"] = charaList
    with open('./chara.json', 'w',encoding="utf-8") as f:
        json.dump(chara, f, ensure_ascii=False)
asyncio.run(get())